File: 3ag/lab01~inpo.cpp
/**
 * Algoritmi genetici — codificare binară
 * Inițializarea unei populații cu parametrii:
 * - n: dimensiunea cromozomului
 * - m: dimensiunea populației
 */

//#define NDEBUG

#include "libcrmz.h"
#include <stdlib.h>
#include <time.h>

#include <iostream>
#include <list>
using namespace std;

int main(void)
{
	int i, n, m;
	list<crmz> pocz;		// populație cromozomi
	time_t t;

	srand(time(&t));
	cout << "Introduceți dimeniunea cromozomului: ";
	cin >> n;
	cout << "Introduceți dimeniunea populației: ";
	cin >> m;

	// inițializare populație
	for (i = 0; i < m; ++i)
		pocz.push_back(crmz(n,i));

	// afișare populație
	cout << endl
	     << "Ce cromozomi au fost generați automat?" << endl;
	for (list<crmz>::iterator icz = pocz.begin(); icz != pocz.end(); ++icz) {
		icz->init();
		cout << icz->getId() << ": " << *icz << endl;
	}
	return 0;
}

File: 3ag/lab01~inpo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Acesta este un program test de generare aleatoare a unei
populații de cromozomi în codificare binară.
"""

import crmz


def main():
    try:
        n = int(input('Introduceți dimeniunea cromozomului: '))
        m = int(input('Introduceți dimeniunea populației: '))
    except ValueError:
        print('Sigur ați introdus două numere?')
        raise
    except:
        print('A apărut o excepție neașteptată...')
        raise

    popcr = []
    # inițializare populație
    for i in range(m):
        popcr.append(crmz.CrmzCB(n))

    # afișare populație
    print('')
    print('Ce cromozomi în codificare binară au fost generați automat?');
    for cr in popcr:
        print('%d: %s' % (cr.id, cr.sir()))


if __name__ == '__main__':
    main()
